utils/s3/aws_errors.cc

```diff
@@ -0,0 +1,136 @@
+    {"ServiceUnavailableException", aws_error(aws_error_type::SERVICE_UNAVAILABLE, retryable::yes)},
+    {"ServiceUnavailableError", aws_error(aws_error_type::SERVICE_UNAVAILABLE, retryable::yes)},
+    {"ServiceUnavailable", aws_error(aws_error_type::SERVICE_UNAVAILABLE, retryable::yes)},
+    {"RequestThrottledException", aws_error(aws_error_type::THROTTLING, retryable::yes)},
+    {"RequestThrottled", aws_error(aws_error_type::THROTTLING, retryable::yes)},
+    {"ThrottlingException", aws_error(aws_error_type::THROTTLING, retryable::yes)},
+    {"ThrottledException", aws_error(aws_error_type::THROTTLING, retryable::yes)},
+    {"Throttling", aws_error(aws_error_type::THROTTLING, retryable::yes)},
+    {"ValidationErrorException", aws_error(aws_error_type::VALIDATION, retryable::no)},
+    {"ValidationException", aws_error(aws_error_type::VALIDATION, retryable::no)},
+    {"ValidationError", aws_error(aws_error_type::VALIDATION, retryable::no)},
+    {"AccessDeniedException", aws_error(aws_error_type::ACCESS_DENIED, retryable::no)},
+    {"AccessDenied", aws_error(aws_error_type::ACCESS_DENIED, retryable::no)},
+    {"ResourceNotFoundException", aws_error(aws_error_type::RESOURCE_NOT_FOUND, retryable::no)},
+    {"ResourceNotFound", aws_error(aws_error_type::RESOURCE_NOT_FOUND, retryable::no)},
+    {"UnrecognizedClientException", aws_error(aws_error_type::UNRECOGNIZED_CLIENT, retryable::no)},
+    {"UnrecognizedClient", aws_error(aws_error_type::UNRECOGNIZED_CLIENT, retryable::no)},
+    {"SlowDownException", aws_error(aws_error_type::SLOW_DOWN, retryable::yes)},
+    {"SlowDown", aws_error(aws_error_type::SLOW_DOWN, retryable::yes)},
+    {"SignatureDoesNotMatchException", aws_error(aws_error_type::SIGNATURE_DOES_NOT_MATCH, retryable::no)},
+    {"SignatureDoesNotMatch", aws_error(aws_error_type::SIGNATURE_DOES_NOT_MATCH, retryable::no)},
+    {"InvalidAccessKeyIdException", aws_error(aws_error_type::INVALID_ACCESS_KEY_ID, retryable::no)},
+    {"InvalidAccessKeyId", aws_error(aws_error_type::INVALID_ACCESS_KEY_ID, retryable::no)},
+    {"RequestTimeTooSkewedException", aws_error(aws_error_type::REQUEST_TIME_TOO_SKEWED, retryable::yes)},
+    {"RequestTimeTooSkewed", aws_error(aws_error_type::REQUEST_TIME_TOO_SKEWED, retryable::yes)},
+    {"RequestTimeoutException", aws_error(aws_error_type::REQUEST_TIMEOUT, retryable::yes)},
+    {"RequestTimeout", aws_error(aws_error_type::REQUEST_TIMEOUT, retryable::yes)}};
+
+aws_error::aws_error(aws_error_type error_type, retryable is_retryable) : _type(error_type), _is_retryable(is_retryable) {
+}
+
+aws_error::aws_error(aws_error_type error_type, std::string&& error_message, retryable is_retryable)
+    : _type(error_type), _message(std::move(error_message)), _is_retryable(is_retryable) {
+}
+
+aws_error aws_error::parse(seastar::sstring&& body) {
+    aws_error ret_val;
+
+    if (body.empty()) {
+        return ret_val;
+    }
+
+    rapidxml::xml_document<> doc;
```

Comment:
This is large on-stack object. We fixed this in other xml parsing code with 37f1beade52ebe589929c588381fd593bf9f0163 . Worth doing the same here